### PR TITLE
ddtrace/tracer: Use DD_AGENT_HOST to set trace agent hostname before querying the trace-agent for its features

### PR DESCRIPTION
**Pull request by carflo**
If the env var `DD_AGENT_HOST` is set and the trace-agent is not running
on localhost, then the call to the agent's `/info` endpoint fails as its
using the wrong hostname.

### CODE REVIEW

**Review by gbbr — APPROVED** [review]
You're right! Thanks!

File: ddtrace/tracer/option.go [hunk 1]
```diff
@@ -190,6 +190,9 @@ func newConfig(opts ...StartOption) *config {
 	if v := os.Getenv("DD_TRACE_SOURCE_HOSTNAME"); v != "" {
 		c.hostname = v
 	}
+	if v := os.Getenv("DD_AGENT_HOST"); v != "" {
+		c.agentAddr = v
+	}
```
Comment (gbbr) [inline ddtrace/tracer/option.go:195]: Actually, looking at the code more, what do you think if we would instead modify this line at the top of `newConfig`:
```go
	c.agentAddr = defaultAddress
```
to become:
```go
	c.agentAddr = resolveAddr(defaultAddress)
```

`resolveAddr` is in `transport.go` and should likely be renamed more accurately to maybe `resolveAgentAddr`.

Comment (carflo) [reply ddtrace/tracer/option.go]: Great suggestion. I've made that change and renamed `resolveAddr` to `resolveAgentAddr`.

**Review by gbbr — APPROVED** [review]
Very nice! Many thanks for this improvement!

**Pull request merged**